gems/pending/util/miq_winrm.rb

```diff
@@ -17,17 +18,33 @@ def connect(options = {})
     validate_options(options)
     @uri        = build_uri
     @connection = raw_connect(@username, @password, @uri)
+    @operations = 0
     @executor   = @connection.create_executor
   end
 
+  def close
+    @executor.close
+    @executor = nil
+  end
```

Comment:
This will remove the executor, making `self` completely unusable.  That is, if someone calls any method after close, then :boom: .  I assume it's ok, but maybe we want to expose a proper block that autocloses much like `File.open`?  Another option is to have the `executor` method auto-create itself, and not create it in the initializer.  This way, if someone calls close, and then wants to reuse the MiqWinRm object, they can.

So executor would look something like:

```ruby
   def executor
     @operations += 1
     @executor ||= @connection.create_executor
     return @executor if @operations < MAX_WINRM_CMDS
     $log.debug "MiqWinRM: #{@operations} operations on executor - restarting."
     @operations = 1
     close
     executor
   end
```

Btw, the block autoclosing thing I mentioned would be something like...

```ruby
def self.open(options = {})
  handle = new(options)
  return handle unless block_given?

  yield handle
  handle.close
end
```